controllers/csiaddons/csiaddonsnode_controller.go

```diff
@@ -184,6 +196,59 @@ func (r *CSIAddonsNodeReconciler) removeFinalizer(
 	return nil
 }
 
+// resolveEndpoint parses the endpoint and returned a string that can be used
+// by GRPC to connect to the sidecar.
+func (r *CSIAddonsNodeReconciler) resolveEndpoint(ctx context.Context, rawURL string) (string, error) {
+	namespace, podname, port, err := parseEndpoint(rawURL)
+	if err != nil && errors.Is(err, errLegacyEndpoint) {
+		return rawURL, nil
+	} else if err != nil {
+		return "", err
+	}
+
+	pod := &corev1.Pod{}
+	err = r.Client.Get(ctx, client.ObjectKey{
+		Namespace: namespace,
+		Name:      podname,
+	}, pod)
+	if err != nil {
+		return "", fmt.Errorf("failed to get pod %s/%s: %w", namespace, podname, err)
+	}
+
```

Comment:
add a check for `pod.Status.PodIP` here to ensure the pod IP is set?

Reply:
added a check for this as well